core/internal/testutils/testutils.go

```diff
@@ -217,7 +217,6 @@ func (ts *testWSServer) newWSHandler(chainID *big.Int, callback JSONRPCHandler)
 		defer conn.Close()
 		ts.mu.Lock()
 		if ts.wsconns == nil {
-			ts.t.Log("Server closed")
```

Comment:
This can happen after the test is over b/c we haven't called `ts.wg.Add` yet.
Removing this line improves things, but now I'm thinking there is a cleaner more fundamental fix here that would be better.

Reply:
Reordering should work here, with the tradeoff of holding the lock for longer.